prisma/migrations/20230829142201_add_model_version_monetization_table/migration.sql

```diff
@@ -0,0 +1,39 @@
+-- CreateEnum
+CREATE TYPE "ModelVersionSponsorshipSettingsType" AS ENUM ('FixedPrice', 'Bidding');
+
+-- CreateEnum
+CREATE TYPE "ModelVersionMonetizationType" AS ENUM ('PaidAccess', 'PaidEarlyAccess', 'CivitaiClubOnly', 'MySubscribersOnly', 'Sponsored');
+
+-- CreateTable
+CREATE TABLE "ModelVersionSponsorshipSettings" (
+    "id" SERIAL NOT NULL,
+    "modelVersionMonetizationId" INTEGER NOT NULL,
+    "type" "ModelVersionSponsorshipSettingsType" NOT NULL DEFAULT 'FixedPrice',
+    "currency" TEXT NOT NULL,
+    "unitAmount" INTEGER NOT NULL,
+
+    CONSTRAINT "ModelVersionSponsorshipSettings_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "ModelVersionMonetization" (
+    "id" SERIAL NOT NULL,
+    "modelVersionId" INTEGER NOT NULL,
+    "type" "ModelVersionMonetizationType" NOT NULL DEFAULT 'PaidAccess',
+    "currency" TEXT NOT NULL,
+    "unitAmount" INTEGER,
+
+    CONSTRAINT "ModelVersionMonetization_pkey" PRIMARY KEY ("id")
+);
```

Comment:
This is more complex than I would have thought to do at this point. Mind explaining what made you decided to do it this way?

Reply:
Also, you have a `currency` here but nothing corresponding to that in the UI.

Reply:
> Also, you have a currency here but nothing corresponding to that in the UI.

Currency is defaulted to USD when creating the actual record via the service, added the field in case we do indeed want to support it. 

>This is more complex than I would have thought to do at this point. Mind explaining what made you decided to do it this way?

Rather than add 3 new fields to the modelVersion table which we'd need to delete further down the line (`monetizationType / sponsorshipType / sponsorshipValue`), I thought it made sense to just go with what I think would scale somewhat better once we started building on it. Tho I'm still unclear on how things will evolve and what we will support later on (i.e, do we want multiple monetization types?) I thought this is just barebones enough that we can build on it but not big enough that we can't change it. Appeared a good middleground to me.